lib/matplotlib/spines.py

```diff
@@ -452,6 +452,9 @@ def linear_spine(cls, axes, spine_type, **kwargs):
         else:
             raise ValueError('unable to make path for spine "%s"' % spine_type)
         result = cls(axes, spine_type, path, **kwargs)
+        if not rcParams['axes.spines.{0}'.format(spine_type)]:
+            result.set_visible(False)
```

Comment:
I would prefer this as `result.set_visible(rcParams['axes.spines.{0}'.format(spine_type)])`